qiskit/quantum_info/operators/measures.py

```diff
@@ -375,3 +387,28 @@ def _input_formatter(obj, fallback_class, func_name, arg_name):
     warnings.warn(
         'Treating array input as a {} object'.format(fallback_class.__name__))
     return fallback_class(obj)
+
+
+def _cp_condition(channel):
+    """Return Choi-matrix eigenvalues for checking if channel is CP"""
+    if isinstance(channel, QuantumChannel):
+        if not isinstance(channel, Choi):
+            channel = Choi(channel)
+        return np.linalg.eigvalsh(channel.data)
+    unitary = Operator(channel).data
+    return np.tensordot(unitary, unitary.conj(), axes=([0, 1], [0, 1])).real
+
+
+def _tp_condition(channel):
+    """Return partial tr Choi-matrix eigenvalues for checking if channel is TP"""
+    if isinstance(channel, QuantumChannel):
+        if not isinstance(channel, Choi):
+            channel = Choi(channel)
+        choi = channel.data
+        dims = tuple(np.sqrt(choi.shape).astype(np.int))
+        shape = dims + dims
+        tr_choi = np.trace(np.reshape(choi, shape), axis1=1, axis2=3)
+    else:
+        unitary = Operator(channel).data
+        tr_choi = np.tensordot(unitary, unitary.conj(), axes=(0, 0))
+    return np.linalg.eigvalsh(tr_choi - np.eye(len(tr_choi)))
```

Comment:
In the `is_tp` method this uses `allclose` method to evaluate if the trace is identity. Is there any numerical difference between these methods?

Reply:
All close doesn't allow you to quantify how close to trace preserving a channel is. This is more accurate way of quantifying TP (if evals < 0 trace decreasing, and > 0 trace increasing).

Reply:
Okey, so we need to update `is_tp` method in another PR (or just write an issue)? I think `is_*` methods and this function should return the same result regardless of input `chan`.

Reply:
The channel `is_tp` implementation could (and probably should) be updated to check this way, but the end user behavior of that function won't change since it only returns True/False.